lib/fluent/plugin/parser_syslog.rb

```diff
@@ -137,6 +153,91 @@ def parse_plain(text, &block)
 
         yield time, record
       end
+
+      SPLIT_CHAR = ' '.freeze
+      PRI_START_CHAR = '<'.freeze
+
+      def parse_rfc3164(text, &block)
+        pri = nil
+        start = 0
+        if @with_priority
+          if text.start_with?(PRI_START_CHAR)
+            i = text.index('>'.freeze, 1)
+            pri = text.slice(1, i - 1).to_i
```

Comment:
I don't think so because the purpose of this parser is for supporting more format, not strict parser.

Reply:
> I don't think so because the purpose of this parser is for supporting more format, not strict parser.

What kind of format do we want to support? 
If this parser does not follow the rfc3164, it's probably better to use new `message_format ` name (rfc3164_ext or like that).

Reply:
> If this parser does not follow the rfc3164, it's probably better to use new message_format name (rfc3164_ext or like that).

The problem is existing products uses `rfc3164` for it. rfc3164 describes the collection of existing message format and BSD spec but many existing tools doesn't follow rfc3164 strictly. This parser can support strict rfc3164 format, so it doesn't block user's usecase for me.

Reply:
> pri can be 0 if i < 2 or text.slice(1, i - 1) contains not number charactors. Is it acceptable?

Is it okay?